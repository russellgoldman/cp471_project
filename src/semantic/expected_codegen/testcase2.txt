num1 = 10
num2 = 5
num1 = num1 + num2
if num1 > 14 goto L2
goto L3
L2: num2 = None + 1
goto L4
L3: num2 = None + 4
L4: