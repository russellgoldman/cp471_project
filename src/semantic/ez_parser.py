# Ref: https://ply.readthedocs.io/en/latest/ply.html
import ply.yacc as yacc
import grammar_rules_ast
from lexer import Lexer

class Parser(object):
    # Build the parser using the grammar rules module
    def build(self, **kwargs):
        self.lexer = Lexer()
        self.lexer.build()
        self.parser = yacc.yacc(module=grammar_rules_ast, **kwargs)

    def parse(self, data, debug=False):
        return self.parser.parse(data, debug=debug, lexer = self.lexer)