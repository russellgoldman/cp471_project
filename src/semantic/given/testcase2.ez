Number num1 = 10;
Number num2 = 5;
num1 = num1 + num2;
if(num1 > 14) {
    num2 = 10 / 2 + 1;
}
else {
    num2 = 2 * 3 + 4;
}
