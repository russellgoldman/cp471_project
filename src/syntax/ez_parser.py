# Ref: https://ply.readthedocs.io/en/latest/ply.html
import sys, os

sys.path.append(os.path.abspath(os.path.join('..')))
import ply.yacc as yacc
import grammar_rules
from lexical.lexer import Lexer

class Parser(object):
    # Build the parser using the grammar rules module
    def build(self, **kwargs):
        self.lexer = Lexer()
        self.lexer.build()
        self.parser = yacc.yacc(module=grammar_rules, **kwargs)

    def parse(self, data, debug=False):
        return self.parser.parse(data, debug=debug, lexer = self.lexer)