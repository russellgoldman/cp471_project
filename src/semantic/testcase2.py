import unittest
import sys, os
sys.path.append(os.path.abspath(os.path.join('..', 'utilities')))

from ez_parser import Parser

class TestCase2(unittest.TestCase):
    def test_bedmas_operations(self) -> None:
        source_f = open('./given/testcase2.ez', 'r')
        source = source_f.read()
        source_f.close()

        expected_f = open('./expected/testcase2.txt', 'r')
        expected = expected_f.read()
        expected_f.close()

        # build parser
        p = Parser()
        p.build()
        # generate parse tree
        given_tree = p.parse(source)
        self.assertEqual(str(given_tree), expected)


if __name__ == '__main__':
    unittest.main()