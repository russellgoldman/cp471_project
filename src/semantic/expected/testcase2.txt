('statement', ('assignmentExpression', '=', 'num1', 10), ('statement', ('assignmentExpression', '=', 'num2', 5), ('statement', ('assignmentExpression', '=', 'num1', ('sumExpression', '+', 'num1', 'num2')), ('ifStatement', ('relationExpression', '>', 'num1', 14), ('assignmentExpression', '=', 'num2', ('sumExpression', '+', ('multiplyExpression', '/', 10, 2), 1)), ('assignmentExpression', '=', 'num2', ('sumExpression', '+', ('multiplyExpression', '*', 2, 3), 4))))))